extensions/azure/processors/PutAzureBlobStorage.h

```diff
@@ -0,0 +1,136 @@
+
+class PutAzureBlobStorageTestsFixture;
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace azure {
+namespace processors {
+
+class PutAzureBlobStorage : public core::Processor {
+ public:
+  static constexpr char const* ProcessorName = "PutAzureBlobStorage";
+
+  // Supported Properties
+  static const core::Property ContainerName;
+  static const core::Property AzureStorageCredentialsService;
+  static const core::Property StorageAccountName;
+  static const core::Property StorageAccountKey;
+  static const core::Property SASToken;
+  static const core::Property CommonStorageAccountEndpointSuffix;
+  static const core::Property ConnectionString;
+  static const core::Property Blob;
+  static const core::Property CreateContainer;
+
+  // Supported Relationships
+  static const core::Relationship Failure;
+  static const core::Relationship Success;
+
+  explicit PutAzureBlobStorage(const std::string& name, const minifi::utils::Identifier& uuid = minifi::utils::Identifier())
+    : PutAzureBlobStorage(name, uuid, nullptr) {
+  }
+
+  ~PutAzureBlobStorage() override = default;
+
+  void initialize() override;
+  void onSchedule(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSessionFactory> &sessionFactory) override;
+  void onTrigger(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSession> &session) override;
+
+  class ReadCallback : public InputStreamCallback {
+   public:
+    ReadCallback(uint64_t flow_size, azure::storage::BlobStorage& blob_storage_wrapper, const std::string &blob_name)
+      : flow_size_(flow_size)
+      , blob_storage_wrapper_(blob_storage_wrapper)
+      , blob_name_(blob_name) {
+    }
+
+    int64_t process(const std::shared_ptr<io::BaseStream>& stream) override {
+      std::vector<uint8_t> buffer;
+      int read_ret = stream->read(buffer, flow_size_);
+      if (read_ret < 0) {
+        return -1;
+      }
+
+      result_ = blob_storage_wrapper_.uploadBlob(blob_name_, buffer.data(), flow_size_);
+      if (!result_) {
+        return -1;
+      }
+      return result_->length;
+    }
+
+    utils::optional<azure::storage::UploadBlobResult> getResult() const {
+      return result_;
+    }
+
+   private:
+    uint64_t flow_size_;
+    azure::storage::BlobStorage &blob_storage_wrapper_;
+    std::string blob_name_;
+    utils::optional<azure::storage::UploadBlobResult> result_ = utils::nullopt;
+  };
+
+ private:
+  friend class ::PutAzureBlobStorageTestsFixture;
+
+  explicit PutAzureBlobStorage(const std::string& name, const minifi::utils::Identifier& uuid, std::unique_ptr<storage::BlobStorage> blob_storage_wrapper)
+    : core::Processor(name, uuid)
+    , blob_storage_wrapper_(std::move(blob_storage_wrapper)) {
+  }
+
+  std::string getConnectionStringFromControllerService(const std::shared_ptr<core::ProcessContext> &context) const;
+  std::string getAzureConnectionStringFromProperties(
```

Comment:
`getAzureConnectionStringFromProperties` could be made static and possibly hidden in the implementation file.

Reply:
Updated in 1f07ca2f5b54c5ba7f4975634e9e320c1c0c9449